api/webhook.go

```diff
@@ -550,3 +562,71 @@ func publishToQueue(queue queue.Service, p *pipeline.Build, b *library.Build, r
 		}
 	}
 }
+
+// renameRepository is a helper function that takes the old name of the repo,
+// queries the database for the repo that matches that name and org, and updates
+// that repo to its new name in order to preserve it. It also updates the secrets
+// associated with that repo.
+func renameRepository(h *library.Hook, r *library.Repo, c *gin.Context) error {
+	// get the old name of the repo
+	previousName := r.GetPreviousName()
+	// get the repo from the database that matches the old name
+	dbR, err := database.FromContext(c).GetRepo(r.GetOrg(), previousName)
+	if err != nil {
+		retErr := fmt.Errorf("%s: failed to get repo from database", baseErr)
+		util.HandleError(c, http.StatusBadRequest, retErr)
+		h.SetStatus(constants.StatusFailure)
+		h.SetError(retErr.Error())
+		return retErr
+	}
+
+	// update the repo name information
+	dbR.SetName(r.GetName())
+	dbR.SetFullName(r.GetFullName())
+	dbR.SetClone(r.GetClone())
+	dbR.SetLink(r.GetLink())
+	dbR.SetPreviousName(previousName)
+
+	// update the repo in the database
+	err = database.FromContext(c).UpdateRepo(dbR)
+	if err != nil {
+		retErr := fmt.Errorf("%s: failed to update repo in database", baseErr)
+		util.HandleError(c, http.StatusBadRequest, retErr)
+		h.SetStatus(constants.StatusFailure)
+		h.SetError(retErr.Error())
+		return retErr
+	}
+
+	// get total number of secrets associated with repository
+	// nolint: lll // ignore long line due to extensive function arguments
+	t, err := database.FromContext(c).GetTypeSecretCount(constants.SecretRepo, r.GetOrg(), previousName, []string{})
+	if err != nil {
+		return fmt.Errorf("unable to get secrets for repo %s: %w", previousName, err)
```

Comment:
Could we include the `org` in this error message?

```suggestion
		return fmt.Errorf("unable to get secrets for repo %s/%s: %w", r.GetOrg(), previousName, err)
```